turgon gtest fail on ubuntu 18.04

I found that the turgon's google test(gtest) cannot pass on ubuntu 18.04, because of g++ syntax error:

> In file included from .../turgon/spacetime/include/spacetime/ElementBase.hpp:9:0,                                                                                                    
                 from .../turgon/spacetime/include/spacetime.hpp:13,                                                                                                                 
                 from .../turgon/spacetime/gtests/main.cpp:3:                                                                                                                        
.../turgon/spacetime/include/spacetime/Field_decl.hpp:32:29: error: ‘function’ in namespace ‘std’ does not name a template type                                                      
     using calc_type1 = std::function<value_type(Selm const &, size_t)>;                                                                                                                                    
                             ^~~~~~~~                                                                                                                                                                       
.../turgon/spacetime/include/spacetime/Field_decl.hpp:33:29: error: ‘function’ in namespace ‘std’ does not name a template type                                                      
     using calc_type2 = std::function<void (Selm &)>;                                                                                                                                                       
                             ^~~~~~~~                                                                                                                                                                       
.../turgon/spacetime/include/spacetime/Field_decl.hpp:45:5: error: ‘calc_type1’ does not name a type; did you mean ‘value_type’?                                                     
     calc_type1 const & xn_calc() const { return m_xn_calc; }                                                                                                                                               
     ^~~~~~~~~~                                                                                                                                                                                             
     value_type                                                                                                                                                                                             
.
.
.
.

 It seems to be missing`#include <functional>` in ` turgon/spacetime/include/spacetime/Field_decl.hpp`, but I don't  have same problem on macos.
I also try to add  `#include <functional>` into ` turgon/spacetime/include/spacetime/Field_decl.hpp`, then the gtest will pass.

--- a/spacetime/include/spacetime/Field_decl.hpp
+++ b/spacetime/include/spacetime/Field_decl.hpp
@@ -7,6 +7,7 @@
 
 #include <memory>
 #include <vector>
+#include <functional>
 
 #include "spacetime/system.hpp"
 #include "spacetime/type.hpp"

